MigrationGuide.md

```diff
@@ -0,0 +1,412 @@
+
+Below is an example of usage that would have worked prior to these changes but is now likely to cause an application to error:
+
+```swift
+let query = Select(from: myTable)
+connection.execute(query: query) { result in
+    //Handle result
+}
+let newQuery = Select(from: otherTable)
+connection.execute(query: newQuery) { result in
+    //Handle result
+}
+```
+
+Because the execute API now behaves asynchronously the code will immediately return from the first call to connection.execute resulting in two concurrent operations on the same connection, those being executing query and executing newQuery.
+
+To do this properly you would need to nest the calls as in the example below:
+
+```swift
+let query = Select(from: myTable)
+connection.execute(query: query) { result in
+    //Handle result
+    let newQuery = Select(from: otherTable)
+    connection.execute(query: newQuery) { result in
+        //Handle result
+    }
+}
+```
+
+In this case the second connection.execute call is run after the first call has completed by virtue of it being part of the callback from the first call.
+
+This principal applies equally to all the asynchronous API’s in Swift Kuery and is important for the project as it will allow easier adoption of an alternative database driver in future, for example one which uses the asynchronous c api’s or perhaps a pure swift driver.
+
+## Getting a database connection
+
+The connection protocol’s connect method has been updated to accept a callback which will be called in an asynchronous fashion, previously the call would have behaved synchronously. The updated method will pass a QueryResult to the callback that will indicate whether the connection was successfully made or not.
+
+Prior to this change your code for creating and using a connection would have looked similar to the example below:
+
+```swift
+let connection = PostgreSQLConnection(….)
+// Connect the connection
+connection.connect { error in
+    if let error = error {
+        // Connection not established - handle and return
+     }
+     // Connection established
+     connection.execute(….) { result in
+         // Handle result
+    }
+}
+```
+
+With the new method your code will need to be updated to handle the change in return type in a manner similar to that in the example below:
+
+```swift
+let connection = PostgreSQLConnection(….)
+// Connect the connection
+connection.connect { queryResult in
+    guard let _ = queryResult.success else {
+        // Connection not established, call queryResult.asError to get error.
+        return
+    }
+     // Connection established
+     connection.execute(….) { result in
+         // Handle result
+    }
+}
+```
+
+The connection protocol has also been updated to include a connectSync method. As suggested by it’s name this function is synchronous in nature, An example of its usage is below:
```

Comment:
Can you put back-ticks around the key function names (connectSync)
Also, `it's` -> `its`